src/dev-app/theme-token-api.scss

```diff
@@ -16,33 +18,47 @@ dev-app {
   }
 }
 
-.demo-unicorn-dark-theme {
-  background: black;
-  color: white;
-}
-
+// Emit Angular Material core styles.
 @include mat.core();
 
-$theme: mat.define-light-theme((
+// Base theme configuration for our M3 theme.
+$m3-base-config: (
   color: (
-    primary: mat.define-palette(mat.$indigo-palette),
-    accent: mat.define-palette(mat.$pink-palette),
+    primary: matx.$m3-green-palette
   ),
-  typography: mat.define-typography-config(),
-  density: 0,
-));
+  typography: (
+    brand-family: 'monospace'
+  )
+);
+
+// Our M3 light theme.
+$light-theme: matx.define-theme($m3-base-config);
 
-// Apply all tokens (derived from `$theme`) to the `html` element. This ensures that all components
-// on the page will inherit these tokens.
+// Our M3 dark theme.
+$dark-theme: matx.define-theme(map.set($m3-base-config, color, theme-type, dark));
```

Comment:
Is this only for testing or the final API for defining dark themes?

Reply:
It's one possible way someone could do it. To create a dark theme you need to pass a config like this:
```scss
$theme: mat.define-theme((
  color: (
    theme-type: dark,
    ...
  ),
  ...
));
```

I chose to create the config using `map.set` here, but users could hardcode it or use whatever other method they want to come up with the config.

I'm happy to discuss more about the API if you want to set up a meeting.

Reply:
I don't love the current `define-light-theme`, `define-dark-theme` because most of the configuration is encoded in the config object you pass in, but a tiny bit is encoded in the mixin you call.

Reply:
I think the objection is that putting light/dark in the mixin name is specific to just a portion of the theme: color. It made more sense before we included typography and density